oop-relationships/inheritance.md

```diff
@@ -15,15 +15,261 @@
 | Vocab | Definition | Synonyms | How to Use in a Sentence
 | --- | --- | --- | ---
 
-## Syntax: Subclass Inherits from Superclass
+## Syntax for Inheritance
 
-## Good to Know: Everything Inherits from `Object`
+To implement inheritance between two classes, we include the parent class in the definition of the child class:
+
+```python
+class ExampleChildClass(ExampleParentClass):
+    pass
+```
+
+Compared to classes without explicit inheritance, this syntax:
+
+- Adds `()`s to the right of the class name
+- Puts the name of the parent class in the `()`s
```

Comment:
😂 😂 😂 😂  

Off-topic: I love learning about this kind of thing. I don't know what this whole concept is called. Do you know what it's called? I should learn about it formally.

I think I observed this a lot in Rails in places where we could either throw in a class name or an instance of that class, and both worked "as expected"

Reply:
I'm not sure.  It might just be a metaclass thing.  How in Ruby and Python, a class name isn't just a type name (like it is in say C++), but it's just another variable that evaluates to a class object.  And the list of stuff that can go in Pythons derivation syntax is a list of the class objects that constitute the lookup chain (more or less).

I didn't really become aware of this until I was trying to avoid some cyclical imports in flask and wound up stashing a type elsewhere for export... or something to that effect.  It was traumatic, and the details are hazy. 😆